### PR TITLE
Support `DISTINCT ON (...)`

### LINKED ISSUE #806: In correct serialization of `DISTINCT ON()` syntax with postgres
Here is the failing sql: 

```sql
SELECT
   DISTINCT ON (column1) column_alias,
   column2
FROM
   table_name
ORDER BY
   column1,
   column2;
```

Note that this parses without errors because it parses the `ON (column1)` as a function.

When serializing back to string from ast, it produces: 

```sql
SELECT DISTINCT ON(column1) AS column_alias, column2 FROM table_name ORDER BY column1, column2
```

The `AS` should not be there (it is an error in Postgres)

### PR DESCRIPTION
**Pull request by aljazerzen**
Closes #806

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -2300,16 +2300,31 @@ impl<'a> Parser<'a> {
...
+        if all {
+            return parser_err!("Cannot specify both ALL and DISTINCT".to_string());
+        }
+        let on = self.parse_keyword(Keyword::ON);
+        if !on {
+            return Ok(Some(Distinct::Distinct));
+        }
+
+        self.expect_token(&Token::LParen)?;
+        let col_names = if self.consume_token(&Token::RParen) {
```
Comment (alamb) [inline src/parser.rs:2320]: This looks like special case code to handle `SELECT DISTINCT ON() ` - can you please add a test for this case?

**Review by alamb — COMMENTED** [review]
Thank you @aljazerzen  -- this looks very nice. I think it just needs one more test and we could merge it in

File: src/parser.rs (hunk 1 above)
Comment (aljazerzen) [reply src/parser.rs:2320]: :+1:

**Review by alamb — APPROVED** [review]
Looks great -- thank you again @aljazerzen

**Pull request merged**